arwen/elrondapi/elrondei.go

```diff
@@ -1002,6 +1008,27 @@ func getCaller(context unsafe.Pointer, resultOffset int32) {
 	}
 }
 
+//export nonPayableFuncCheck
+func nonPayableFuncCheck(context unsafe.Pointer) {
+	runtime := arwen.GetRuntimeContext(context)
+	metering := arwen.GetMeteringContext(context)
+
+	gasToUse := metering.GasSchedule().ElrondAPICost.GetCallValue
+	metering.UseGas(gasToUse)
+
+	vmInput := runtime.GetVMInput()
+	if vmInput.CallValue.Sign() > 0 {
```

Comment:
vmInput.CallValue.Cmp(zero) > 0 as with ESDTValue -> but for esdt check for nil first.